eclair-core/src/main/scala/fr/acinq/eclair/channel/fund/InteractiveTxBuilder.scala

```diff
@@ -766,16 +766,30 @@ private class InteractiveTxBuilder(replyTo: ActorRef[InteractiveTxBuilder.Respon
   }
 
   private def signTx(unsignedTx: SharedTransaction): Unit = {
+    import fr.acinq.bitcoin.scalacompat.KotlinUtils._
+
     val tx = unsignedTx.buildUnsignedTx()
     val sharedSig_opt = fundingParams.sharedInput_opt.map(_.sign(keyManager, channelParams, tx))
     if (unsignedTx.localInputs.isEmpty) {
       context.self ! SignTransactionResult(PartiallySignedSharedTransaction(unsignedTx, TxSignatures(fundingParams.channelId, tx, Nil, sharedSig_opt)))
     } else {
-      context.pipeToSelf(wallet.signTransaction(tx, allowIncomplete = true).map {
-        case SignTransactionResponse(signedTx, _) =>
+      // we only sign our wallet inputs, and check that we can spend our wallet outputs
+      val ourWalletInputs = unsignedTx.localInputs.map(i => tx.txIn.indexWhere(_.outPoint == OutPoint(i.previousTx, i.previousTxOutput.toInt)))
+      val ourWalletOutputs = unsignedTx.localOutputs.map(o => tx.txOut.indexWhere(output => output.amount == o.amount && output.publicKeyScript == o.pubkeyScript))
```

Comment:
The `localOutputs` can contain either `Local.Change` or `Local.NonChange` (splice-out). If you use all of them here, that means we restrict splice-out to only send to our own wallet. Is that what we want? I believe users will want to be able to splice-out to external addresses.

We definitely want our node to have this restriction, we should make this configurable? We should add to `eclair.conf` a boolean `allow-splice-out-external-addresses` and tweak our behavior here based on this flag.

Reply:
I think that skipping verification of `Local.NonChange` should work if we assume that such outputs have been set by trusted sources (like a secure API call), see https://github.com/ACINQ/eclair/pull/2613/commits/5b4572e8a04de991fe090dc3ceadd813deb0c69a, wdyt ?

Reply:
That sounds reasonable indeed and should work for our current use-cases :+1: